rewriter/src/main/java/com/facebook/coresql/rewriter/OrderByRewriter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.rewriter;
+
+import com.facebook.coresql.parser.AstNode;
+import com.facebook.coresql.parser.OrderByClause;
+import com.facebook.coresql.parser.Unparser;
+import com.facebook.coresql.warning.WarningCollector;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.facebook.coresql.warning.StandardWarningCode.REWRITE_RULE_APPLIED;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class OrderByRewriter
+        extends Rewriter
+{
+    public Set<AstNode> patternMatchedNodes;
+
+    public OrderByRewriter(WarningCollector collector)
+    {
+        super(collector);
+        this.patternMatchedNodes = new HashSet<>();
+    }
+
+    @Override
+    public String rewrite(AstNode node)
+    {
+        requireNonNull(node, "AST passed to rewriter was null");
+        setPatternMatchedNodes(new OrderByPatternMatcher().findMatchedNodes(node));
+        return Unparser.unparse(node, this);
+    }
+
+    @Override
+    public void visit(OrderByClause node, Void data)
+    {
+        if (patternMatchedNodes.contains(node)) {
+            super.addToCollector(REWRITE_RULE_APPLIED.getWarningCode(), format("Rewrite applied at: %s", node.getLocation()), node);
```

Comment:
Also have a name of the rewriter and include it in the warning.

Reply:
Will address in a future PR.